lib/core/src/sdk.rs

```diff
@@ -515,10 +549,70 @@ impl LiquidSdk {
             }
 
             // The lockup tx is in the mempool and we accept 0-conf => try to claim
-            // TODO Add 0-conf preconditions check: https://github.com/breez/breez-liquid-sdk/issues/187
+            // Execute 0-conf preconditions check: https://docs.boltz.exchange/v/api/0-conf
             Ok(RevSwapStates::TransactionMempool
             // The lockup tx is confirmed => try to claim
             | RevSwapStates::TransactionConfirmed) => {
+                let history = self.get_swap_script_history(Swap::Receive(receive_swap.clone()))?;
+
+                let Some(lockup_tx_history) = history.first() else {
+                    return Err(anyhow!("Could not find lockup transaction in the mempool"));
+                };
+
+                let electrum_client = self.config.get_electrum_client()?;
+                let transactions = electrum_client.get_transactions(&[lockup_tx_history.txid])?;
+                let Some(lockup_tx) = transactions.first() else {
+                    return Err(anyhow!("Could not find lockup transaction in the mempool"));
+                };
+
+                if swap_state == RevSwapStates::TransactionMempool.to_string() {
+                    // If the amount is greater than the zero-conf limit
+                    let inputs_sum: u64 = lockup_tx
+                        .input
+                        .iter()
+                        .map(|tx_in| tx_in.pegin_data().map(|data| data.value).unwrap_or(0))
+                        .sum();
+
+                    // TODO Make limit user-definable through config
+                    // TODO If wrong amount is broadcast by Boltz, start negative flow (do not
+                    // broadcast claim, even when confirmed)
+                    if inputs_sum > LIQUID_SDK_ZERO_CONF_LIMIT_SAT {
+                        debug!("Amount is too high to claim with zero-conf ({inputs_sum} sat > {LIQUID_SDK_ZERO_CONF_LIMIT_SAT} sat). Waiting for confirmation...");
+                        return Ok(());
+                    }
+
+                    // If the transaction has RBF, see https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki
+                    let rbf_explicit = lockup_tx
+                        .input
+                        .iter()
+                        .any(|input| input.sequence.is_rbf());
+                    let rbf_implicit = lockup_tx_history.height < 0;
+
+                    if rbf_explicit || rbf_implicit {
+                        debug!("Lockup transaction signals RBF. Waiting for confirmation...");
+                        return Ok(());
+                    }
+
+                    // If the fees are higher than our estimated value
+                    let tx_fees: u64 = lockup_tx
+                        .all_fees()
+                        .values()
+                        .sum();
+                    let lower_bound_estimated_fees = lockup_tx.vsize() as f32 * LIQUID_SDK_ZERO_CONF_FEE_RATE / 1000.0 * 0.8;
```

Comment:
Doesn'tn boltz send lowball here? Does this condition allow it?

Reply:
Agreed, we should probably have two cases for mainnet/testnet for now, one with 0.1 sat/vbyte and another for 0.01